pkg/planner/core/exhaust_physical_plans.go

```diff
@@ -1379,10 +1403,115 @@ func (p *LogicalJoin) constructInnerIndexScanTask(
 	}
 	finalStats := ds.tableStats.ScaleByExpectCnt(rowCount)
 	is.addPushedDownSelection(cop, ds, tmpPath, finalStats)
-	t := cop.ConvertToRootTask(ds.SCtx()).(*RootTask)
-	reader := t.GetPlan()
-	t.SetPlan(p.constructInnerByWrapper(wrapper, reader))
-	return t
+	return p.constructIndexJoinInnerSideTask(cop, ds, path, wrapper)
+}
+
+// construct the inner join task by inner child plan tree
+// The Logical include two parts: logicalplan->physicalplan, physicalplan->task
+// Step1: whether agg can be pushed down to coprocessor
+//
+//	Step1.1: If the agg can be pushded down to coprocessor, we will build a copTask and attach the agg to the copTask
+//	There are two kinds of agg: stream agg and hash agg. Stream agg depends on some conditions, such as the group by cols
+//
+// Step2: build other inner plan node to task
+func (p *LogicalJoin) constructIndexJoinInnerSideTask(dsCopTask *CopTask, ds *DataSource, path *util.AccessPath, wrapper *indexJoinInnerChildWrapper) base.Task {
```

Comment:
I think this method name can be improved.
1. It's too similar to the `constructInnerIndexScanTask` above.
2. This method is mostly about `Aggregate` but the name doesn't reflect this.